### DISCUSSION TITLE
Using TCA for non-views

### DISCUSSION THREAD

**Discussion by GreatApe** [pointfreeco/swift-composable-architecture#1333] [Ideas]
Are there any provisions for using TCA's Reducers, Stores etc for non visual features such as my audio controller, which also happens to be a dependency in my traditional SwiftUI TCA app? 

I'm a long time user of redux like architectures (for both apps and non visual features), but new to TCA, and it seems to me that the idea with TCA is that the `Store` is interacted with through a `ViewStore`, you can't call `send` on a `Store` directly etc. So if that is true, and `ViewStore` lives up to its name, perhaps I would need to make my own "`ComponentStore`" or something?

**Comment by NFulkerson** [comment]
I'm having difficulty finding the other discussion about this, but as I recall, `ViewStore` should be seen as a "view into a store" rather than "a store for a view." As such it's completely valid to have state, action, and reducers that operate independently of any view.

**Comment by stephencelis** [comment]
What @NFulkerson said is totally correct, however it should be noted that TCA is optimized for UIs at the moment, due to its thread model and how its concurrency-based APIs are only compatible with reducers/stores that run on the main thread. So it is likely you'll want these non-view reducers/stores to also run on the main thread.

We hope to relax this constraint in the future, but such a change will almost certainly be breaking, so we're taking care before moving ahead with that direction.

**Reply by GreatApe** [reply]
Thanks for the informative reply!
